packages/postgraphile-core/__tests__/unit/read-cache.test.js

```diff
@@ -67,22 +45,109 @@ test("when no readCache flag, persistentMemoizeWithKey should be undefined", asy
+    // call our method and test output
+    const output = await getPostGraphileBuilder({}, [], {
       readCache: "path/to/cache",
     });
-  } catch (e) {
-    error = e;
-  }
-  expect(error).toMatchSnapshot();
-  expect(fs.readFile).toHaveBeenCalledTimes(1);
+    expect(output).toBe(expectedOutput);
+    expect(fs.readFile).toHaveBeenCalledTimes(1);
+    expect(graphileBuild.getBuilder).toHaveBeenCalledTimes(1);
+    // check persistentMemoizeWithKey, the actual "result" of readCache flag
+    const {
+      persistentMemoizeWithKey,
+    } = graphileBuild.getBuilder.mock.calls[0][1];
+    expect(typeof persistentMemoizeWithKey).toBe("function");
+    expect(persistentMemoizeWithKey("__test")).toEqual(true);
+    expect(() => persistentMemoizeWithKey("unknown_key")).toThrow();
+  });
+
+  test("if cache file has invalid content, getPostGraphileBuilder should error", async () => {
+    // mock fs.readFile to return an object
+    fs.readFile.mockImplementationOnce((path, options, cb) =>
+      cb(null, "thisisnotjson")
+    );
+
+    // call our method and check error
+
+    let error;
+    try {
+      await getPostGraphileBuilder({}, [], {
+        readCache: "path/to/cache",
+      });
+    } catch (e) {
+      error = e;
+    }
+    expect(error).toMatchSnapshot();
+    expect(fs.readFile).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe("When readCache is Object", () => {
+  test("persistentMemoizeWithKey should be a valid function", async () => {
+    // mock getBuilder to fake output
+    const expectedOutput = {};
+    graphileBuild.getBuilder.mockResolvedValueOnce(expectedOutput);
+    // call our method and test output
+    const output = await getPostGraphileBuilder({}, [], {
+      readCache: { __test: true },
+    });
+    expect(output).toBe(expectedOutput);
+    expect(graphileBuild.getBuilder).toHaveBeenCalledTimes(1);
+    // check persistentMemoizeWithKey, the actual "result" of readCache flag
+    const {
+      persistentMemoizeWithKey,
+    } = graphileBuild.getBuilder.mock.calls[0][1];
+    expect(typeof persistentMemoizeWithKey).toBe("function");
+    expect(persistentMemoizeWithKey("__test")).toEqual(true);
+    expect(() => persistentMemoizeWithKey("unknown_key")).toThrow();
+  });
+});
+
+describe("when readCache is not String or Object, getPostGraphileBuilder should error", () => {
+  test("when its Boolean", async () => {
+    // call our method with invalid readCache value and check error
+    let error;
+    try {
+      await getPostGraphileBuilder({}, [], {
+        readCache: true,
+      });
+    } catch (e) {
+      error = e;
+    }
+    expect(error).toMatchSnapshot();
+  });
+  test("when its Array", async () => {
+    // call our method with invalid readCache value and check error
+    let error;
+    try {
+      await getPostGraphileBuilder({}, [], {
+        readCache: [],
+      });
+    } catch (e) {
+      error = e;
+    }
+    expect(error).toMatchSnapshot();
```

Comment:
This isn't actually throwing - the error is `undefined`. You might want to add `expect(error).toBeTruthy()` to each of these tests, snapshots should only be used for looking for changes over time (e.g. error message text, new error properties, etc) rather than asserting behaviour.

Reply:
you are right! I'll add that check

Reply:
done